File: app/src/main/java/com/simeo/codexmicromobile/Keycaps.kt
package com.simeo.codexmicromobile

data class KeycapDefinition(
  val id: String,
  val name: String,
  val category: String,
  val description: String
)

object OfficialKeycaps {
  val all = listOf(
    KeycapDefinition("FAST", "高速実行", "アクション", "Fast modeを切り替えます。"),
    KeycapDefinition("APPR", "承認", "アクション", "現在の承認要求を承認します。"),
    KeycapDefinition("REJ", "拒否", "アクション", "現在の承認要求を拒否します。"),
    KeycapDefinition("SPLIT", "分岐", "アクション", "現在のtaskを分岐します。"),
    KeycapDefinition("MIC", "マイク", "アクション", "音声入力を開始します。"),
    KeycapDefinition("CODEX", "送信", "アクション", "composerを送信します。"),
    KeycapDefinition("BUG", "バグ報告", "その他", "feedback画面を開きます。"),
    KeycapDefinition("OAI", "OpenAI", "その他", "OpenAI developer docsを開きます。"),
    KeycapDefinition("TERM", "ターミナル", "開発", "Codex terminalを切り替えます。"),
    KeycapDefinition("DWN", "Markdownコピー", "その他", "会話をMarkdownとしてコピーします。"),
    KeycapDefinition("DEL", "アーカイブ", "アクション", "現在のtaskをarchiveします。"),
    KeycapDefinition("NEW", "新規task", "アクション", "新しいCodex taskを作成します。"),
    KeycapDefinition("NAV", "ブラウザ", "ナビゲーション", "Codex browser tabを開きます。"),
    KeycapDefinition("MAGIC", "Pin切替", "アクション", "現在のtaskのpinを切り替えます。"),
    KeycapDefinition("DIFF", "Review", "開発", "review表示を切り替えます。"),
    KeycapDefinition("PLAY", "環境Action", "開発", "最初の環境アクションを実行します。"),
    KeycapDefinition("GIT", "Git Commit", "開発", "native commit flowを開きます。"),
    KeycapDefinition("BRCH", "Branch Review", "開発", "branch reviewを開きます。"),
    KeycapDefinition("MRG", "Merge Review", "開発", "merge reviewを開きます。"),
    KeycapDefinition("PR", "Pull Request", "開発", "pull-request flowを開きます。"),
    KeycapDefinition("PAINT", "写真追加", "その他", "composerへ写真を追加します。"),
    KeycapDefinition("LAB", "Lab", "その他", "Lab／Settingsを開きます。"),
    KeycapDefinition("PARTY", "Side Chat", "その他", "side chatを開きます。"),
    KeycapDefinition("TIME", "Task管理", "ナビゲーション", "task管理を開きます。"),
    KeycapDefinition("MIND+", "Reasoning +", "アクション", "reasoning effortを上げます。"),
    KeycapDefinition("MIND-", "Reasoning -", "アクション", "reasoning effortを下げます。"),
    KeycapDefinition("SETUP", "設定", "ナビゲーション", "Codex settingsを開きます。"),
    KeycapDefinition("FOLD", "Folder", "ナビゲーション", "folderを開きます。"),
    KeycapDefinition("UPL", "File追加", "その他", "composerへファイルを追加します。"),
    KeycapDefinition("APPS", "Skills", "ナビゲーション", "Codex Skillsを開きます。")
  )

  val ids = all.mapTo(linkedSetOf()) { it.id }
  fun find(id: String) = all.firstOrNull { it.id == id }
}

File: relay/src/keycaps.ts
export const OFFICIAL_KEYCAP_IDS = [
  "FAST", "APPR", "REJ", "SPLIT", "MIC", "CODEX", "BUG", "OAI", "TERM", "DWN",
  "DEL", "NEW", "NAV", "MAGIC", "DIFF", "PLAY", "GIT", "BRCH", "MRG", "PR",
  "PAINT", "LAB", "PARTY", "TIME", "MIND+", "MIND-", "SETUP", "FOLD", "UPL", "APPS"
] as const;

export type OfficialKeycapId = typeof OFFICIAL_KEYCAP_IDS[number];

export const ADDITIONAL_KEYCAPS = [
  { id: "FAST", slug: "fast", name: "Fast Mode", tooltip: "Toggle Fast mode through the native Codex Micro command." },
  { id: "APPR", slug: "approve", name: "Approve", tooltip: "Approve the current Codex request." },
  { id: "REJ", slug: "reject", name: "Reject", tooltip: "Reject the current Codex request." },
  { id: "SPLIT", slug: "split", name: "Fork Chat", tooltip: "Fork the current Codex chat." },
  { id: "NEW", slug: "new-task", name: "New Task", tooltip: "Create a new Codex task through the native command." },
  { id: "MIND+", slug: "reasoning-up", name: "Reasoning Up", tooltip: "Increase the current composer reasoning effort." },
  { id: "MIND-", slug: "reasoning-down", name: "Reasoning Down", tooltip: "Decrease the current composer reasoning effort." },
  { id: "CODEX", slug: "codex", name: "Codex / Submit", tooltip: "Submit the current composer through the native Codex Micro command." },
  { id: "BUG", slug: "bug", name: "Bug / Feedback", tooltip: "Open Codex feedback through the native Codex Micro command." },
  { id: "OAI", slug: "openai-docs", name: "OpenAI Docs", tooltip: "Open the official OpenAI developer documentation." },
  { id: "TERM", slug: "terminal", name: "Terminal", tooltip: "Toggle the Codex terminal." },
  { id: "DWN", slug: "download", name: "Copy Chat Markdown", tooltip: "Copy the current conversation as Markdown." },
  { id: "DEL", slug: "archive", name: "Archive Chat", tooltip: "Archive the current Codex chat." },
  { id: "NAV", slug: "browser", name: "Browser", tooltip: "Open a Codex browser tab." },
  { id: "MAGIC", slug: "pin", name: "Pin / Unpin Chat", tooltip: "Toggle the pinned state of the current chat." },
  { id: "DIFF", slug: "diff", name: "Review", tooltip: "Toggle the Codex review view." },
  { id: "PLAY", slug: "play", name: "Run Environment Action", tooltip: "Run the first configured environment action." },
  { id: "GIT", slug: "git-commit", name: "Git Commit", tooltip: "Open the native Codex Git commit flow." },
  { id: "BRCH", slug: "branch", name: "Branch Review", tooltip: "Open the Codex review view with the branch keycap." },
  { id: "MRG", slug: "merge", name: "Merge Review", tooltip: "Open the Codex review view with the merge keycap." },
  { id: "PR", slug: "pull-request", name: "Create Pull Request", tooltip: "Open the native Codex pull-request flow." },
  { id: "PAINT", slug: "add-photos", name: "Add Photos", tooltip: "Add photos to the current composer." },
  { id: "LAB", slug: "lab", name: "Lab / Settings", tooltip: "Open Codex settings with the lab keycap." },
  { id: "PARTY", slug: "side-chat", name: "Side Chat", tooltip: "Open a Codex side chat." },
  { id: "TIME", slug: "tasks", name: "Manage Tasks", tooltip: "Open Codex task management." },
  { id: "SETUP", slug: "settings", name: "Settings", tooltip: "Open Codex settings." },
  { id: "FOLD", slug: "open-folder", name: "Open Folder", tooltip: "Open a folder in Codex." },
  { id: "UPL", slug: "add-files", name: "Add Files", tooltip: "Add files to the current composer." },
  { id: "APPS", slug: "skills", name: "Skills", tooltip: "Open Codex Skills." }
] as const satisfies readonly { id: OfficialKeycapId; slug: string; name: string; tooltip: string }[];

